### ISSUE TITLE
When configuration is reloaded, close any active settings dialog

### ISSUE THREAD

**Issue by derekriemer** [nvaccess/nvda#6469]
1. Switch from espeak to SAPI5.
2. open voice settings.
3. press nvda+ctrl+r reloading config.
4. press escape.

Expected: when nvda+ctrl+r is pressed, the dialog should be destroyed because it no longer matters.
What actually happens:
the dialog still exists, trying to close it results in this error.
IO - inputCore.InputManager.executeGesture (21:12:50):
Input: kb(laptop):escape
ERROR - unhandled exception (21:12:50):
Traceback (most recent call last):
  File "gui\settingsDialogs.pyc", line 604, in onCancel
AttributeError: 'VoiceSettingsDialog' object has no attribute 'variantList'
IO - inputCore.InputManager.executeGesture (21:12:51):
Input: kb(laptop):NVDA+f1

This means that any active settings dialog should be nuked when config is reloaded or reset to factory defaults, especially if a synth setting is changed.

**Comment by feerrenrut** [comment]
Once you get into this state you can not close the dialog. For this reason it is a p2.

**Comment by Adriani90** [comment]
I get also following error:

```
IO - inputCore.InputManager.executeGesture (11:28:43.115):
Input: kb(laptop):escape
ERROR - unhandled exception (11:28:43.118):
Traceback (most recent call last):
  File "gui\settingsDialogs.pyc", line 613, in onCancel
RuntimeError: wrapped C/C++ object of type GeneralSettingsPanel has been deleted
```

The dialog is not closing. You can press ok or apply, it doesn't matter.

**Comment by CyrilleB79** [comment]
@derekriemer  and @Adriani90 can you still reproduce this issue?
I cannot with NVDA 2021.2rc1. I can see that the voice panel is updated when you press NVDA+ctrl+R.

It does not seem to

**Comment by Adriani90** [comment]
I cannot reproduce this issue in NVDA 2024.4 Beta. Closing as works for me.

**Comment by CyrilleB79** [comment]
Well the issue is not exactly the same. But reloading the configuration while the speech settings dialog is open still give unexpected results.

STR:
1. Switch from OneCore (default synth) to SAPI5.
2. open voice settings.
3. press nvda+ctrl+r reloading config.
4. Look visually at the dialog while tabbing in it
What actually happens:
All the items from the dialog from "Automatic language switching" onward are drawn at the top of the panel, one above the other, whereas they should be one below the other.

Expected:
This may be discussed, but:
* Either the dialog is closed as requested in this PR
* Or the dialog is correctly updated with the new synth and a normal layout
* Or the dialog remains with a normal layout, but with the old irrelevant synth information, as done for any other settings panel.

**Comment by Adriani90** [comment]
I vote for the third option, the settings pannel should return to its normal state as it was before changing the synthesizer.
I am not sure, but actually in WX Python 4.2.2 it should be possible to get rid of the synth selection dialog, put that combo box into the voice setitngs pannel and update the pannel on the fly as the item in the select synthesizer combo box is changed with up and down arrow keys. Could something like this be investigated?

**Comment by CyrilleB79** [comment]
IMO, closing any settings dialog, as requested here, is the better thing to do when reloading the config from disk. Not only for speech settings, but for all panels.

In any case, if you want to reset to disk config, you probably want to discard the previous changes done.

Keeping the settings dialog open while resetting settings causes other strange things such as the settings dialog keeping a profile's name at the top, while the active profile has been exited with `NVDA+control+R`.

**Comment by seanbudd** [comment]
I'm reproducing something similar.

Steps

1. Set oneCore as your synthesizer
1. Open settings dialog
1. Change your synthesizer to espeak
2. press escape to close and discard settings
1. you are unable to escape the dialog without saving settings first. The following error is logged:

```
ERROR - unhandled exception (14:39:48.884) - MainThread (8128):
Traceback (most recent call last):
  File "gui\settingsDialogs.pyc", line 791, in onCancel
  File "gui\settingsDialogs.pyc", line 1215, in onDiscard
  File "gui\settingsDialogs.pyc", line 1589, in onDiscard
AttributeError: 'VoiceSettingsPanel' object has no attribute 'variantList'
```
